packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts

```diff
@@ -0,0 +1,73 @@
+import { getSqlClientOrThrow } from '../../lib/sql/sqlClient';
+import { registerMigration } from './migrationUtils';
+
+// This updates the functions created in packages/lesswrong/server/manualMigrations/2023-01-13-createViewUpdater.ts
+// to handle the case where a table name is already lowercase
+registerMigration({
+  name: "viewUpdaterErrorHandling",
+  dateWritten: "2023-07-28",
+  idempotent: true,
+  action: async () => {
+    const db = getSqlClientOrThrow();
+
+    const sql = `
+    create or replace function refresh_lowercase_views() returns void as
+    $$
+    DECLARE
+        rec record;
+    BEGIN
+      FOR rec IN
+        select table_name,
+          string_agg(format('%I as %s', column_name, column_name), ', ') cols
+        from information_schema."columns"
+        inner join information_schema."tables" using (table_name)
+        where "columns".table_schema = 'public'
+          and table_name not in ('migration_log', 'mongo2pg_lock')
```

Comment:
I'm not sure I fully understand this part, but the important thing is that in the loop below (and the equivalent loop in `remove_lowercase_views`) rec.table_name goes through the _mixed case_ name of every table in the public schema

Reply:
```
Get all tables. Join against all columns in the db and group by tables.
```